--- a/mysys/mf_tempfile.c
+++ b/mysys/mf_tempfile.c
@@ -112,33 +112,33 @@ File create_temp_file(char *to, const ch
 #elif defined(HAVE_MKSTEMP)
   if (!dir && ! (dir =getenv("TMPDIR")))
     dir= DEFAULT_TMPDIR;
-#ifdef O_TMPFILE
-  {
-    static int O_TMPFILE_works= 1;
-
-    if ((MyFlags & MY_TEMPORARY) && O_TMPFILE_works)
-    {
-      /* explictly don't use O_EXCL here has it has a different
-         meaning with O_TMPFILE
-      */
-      if ((file= open(dir, mode | O_TMPFILE | O_CLOEXEC,
-                      S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP)) >= 0)
-      {
-        my_snprintf(to, FN_REFLEN, "%s/#sql/fd=%d", dir, file);
-        file=my_register_filename(file, to, FILE_BY_O_TMPFILE,
-                                  EE_CANTCREATEFILE, MyFlags);
-      }
-      else if (errno == EOPNOTSUPP || errno == EINVAL)
-      {
-        my_printf_error(EE_CANTCREATEFILE, "O_TMPFILE is not supported on %s "
-                        "(disabling future attempts)",
-                        MYF(ME_NOTE | ME_ERROR_LOG_ONLY), dir);
-        O_TMPFILE_works= 0;
-      }
-    }
-  }
-  if (file == -1)
-#endif /* O_TMPFILE */
+//#ifdef O_TMPFILE
+//  {
+//    static int O_TMPFILE_works= 1;
+//
+//    if ((MyFlags & MY_TEMPORARY) && O_TMPFILE_works)
+//    {
+//      /* explictly don't use O_EXCL here has it has a different
+//         meaning with O_TMPFILE
+//      */
+//      if ((file= open(dir, mode | O_TMPFILE | O_CLOEXEC,
+//                      S_IRUSR | S_IWUSR | S_IRGRP | S_IWGRP)) >= 0)
+//      {
+//        my_snprintf(to, FN_REFLEN, "%s/#sql/fd=%d", dir, file);
+//        file=my_register_filename(file, to, FILE_BY_O_TMPFILE,
+//                                  EE_CANTCREATEFILE, MyFlags);
+//      }
+//      else if (errno == EOPNOTSUPP || errno == EINVAL)
+//      {
+//        my_printf_error(EE_CANTCREATEFILE, "O_TMPFILE is not supported on %s "
+//                        "(disabling future attempts)",
+//                        MYF(ME_NOTE | ME_ERROR_LOG_ONLY), dir);
+//        O_TMPFILE_works= 0;
+//      }
+//    }
+//  }
+//  if (file == -1)
+//#endif /* O_TMPFILE */
   {
     char prefix_buff[30];
     uint pfx_len;
