--- storage/innobase/include/ut0ut.h
+++ storage/innobase/include/ut0ut.h
@@ -69,8 +69,7 @@ Created 1/20/1994 Heikki Tuuri
    independent way by using YieldProcessor. */
 # define UT_RELAX_CPU() YieldProcessor()
 #elif defined(__powerpc__) && defined __GLIBC__
-# include <sys/platform/ppc.h>
-# define UT_RELAX_CPU() __ppc_get_timebase()
+# define UT_RELAX_CPU() __builtin_ppc_get_timebase()
 #else
 # define UT_RELAX_CPU() do { \
      volatile int32	volatile_var; \
@@ -88,9 +87,8 @@ Created 1/20/1994 Heikki Tuuri
 #endif
 
 #if defined(HAVE_HMT_PRIORITY_INSTRUCTION)
-# include <sys/platform/ppc.h>
-# define UT_LOW_PRIORITY_CPU() __ppc_set_ppr_low()
-# define UT_RESUME_PRIORITY_CPU() __ppc_set_ppr_med()
+# define UT_LOW_PRIORITY_CPU() __asm__ __volatile__ ("or 1,1,1")
+# define UT_RESUME_PRIORITY_CPU() __asm__ __volatile__ ("or 2,2,2")
 #else
 # define UT_LOW_PRIORITY_CPU() ((void)0)
 # define UT_RESUME_PRIORITY_CPU() ((void)0)
